doctr/datasets/wildreceipt.py

```diff
@@ -0,0 +1,110 @@
+# Copyright (C) 2021-2023, Mindee.
+
+# This program is licensed under the Apache License 2.0.
+# See LICENSE or go to <https://opensource.org/licenses/Apache-2.0> for full license details.
+
+import glob
+import json
+import os
+from pathlib import Path
+from typing import Any, Dict, List, Tuple, Union
+
+import numpy as np
+from PIL import Image
+
+from .datasets import AbstractDataset
+from .utils import convert_target_to_relative, crop_bboxes_from_image
+
+__all__ = ["WILDRECEIPT"]
+
+from ..utils import polygon_to_bbox
+
+
+class WILDRECEIPT(AbstractDataset):
+    """WildReceipt dataset from `"Spatial Dual-Modality Graph Reasoning for Key Information Extraction"
+        <https://arxiv.org/abs/2103.14470v1>`_ |
+    `repository <https://download.openmmlab.com/mmocr/data/wildreceipt.tar>`_.
+
```

Comment:
**Optional**
If we have an image to give a general overview of the dataset would be great

See: 
https://mindee.github.io/doctr/modules/datasets.html
https://github.com/mindee/doctr/blob/f22f6dd4e038082a90bb6a7b78ffe8a356368fc7/doctr/datasets/funsd.py#L24

Reply:
where should I put the image ?

Reply:
@HamzaGbada you can post it here

@odulcy-mindee  Could you upload it please ?

Reply:
![combined_image](https://github.com/mindee/doctr/assets/50714796/e2352a44-26d4-4125-9954-b8e1a43bcd72)

Reply:
@felixT2K @HamzaGbada Here you go:
```
https://doctr-static.mindee.com/models?id=v0.7.0/wildreceipt-dataset.jpg&src=0
```